synapseclient/core/download/download_functions.py

```diff
@@ -161,6 +170,105 @@ async def download_file_entity(
     entity.cacheDir = os.path.dirname(download_path)
 
 
+async def download_file_entity_model(
+    *,
+    download_location: str,
+    file: "File",
+    if_collision: str,
+    submission: str,
+    synapse_client: Optional["Synapse"] = None,
+) -> None:
+    """
+    Download file entity
+
+    Arguments:
+        download_location: The download location
+        entity:           The File object
+        if_collision:      Determines how to handle file collisions.
+                            May be
+
+            - `overwrite.local`
+            - `keep.local`
+            - `keep.both`
+
+        submission:       Access associated files through a submission rather than through an entity.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    # set the initial local state
+    file.path = None
+
+    # check to see if an UNMODIFIED version of the file (since it was last downloaded) already exists
+    # this location could be either in .synapseCache or a user specified location to which the user previously
+    # downloaded the file
+    cached_file_path = client.cache.get(
+        file_handle_id=file.data_file_handle_id, path=download_location
+    )
+
+    # location in .synapseCache where the file would be corresponding to its FileHandleId
+    synapse_cache_location = client.cache.get_cache_dir(
+        file_handle_id=file.data_file_handle_id
+    )
+
+    file_name = (
+        file.file_handle.file_name
+        if cached_file_path is None
+        else os.path.basename(cached_file_path)
```

Comment:
Ah nice. This `else` statement is meant to catch for cases where a user re-named the file right?

Reply:
The `else` in this case is when we have a "cache hit" and the file was already downloaded and present within the cache. 

As for exactly why this was put in place, I'm not certain. It was one of the items in place where this code was taken from, and I didn't want to change this behavior.